core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -499,36 +502,70 @@ trait WorkspaceComponent {
+     * gets the submission stats (last submission failed date, last submission success date, running submission count)
      * for each workspace
      *
      * @param workspaceIds the workspace ids to query for
      * @return WorkspaceSubmissionStats keyed by workspace id
      */
     def listSubmissionSummaryStats(workspaceIds: Seq[UUID]): ReadAction[Map[UUID, WorkspaceSubmissionStats]] = {
-      // workflow date query: select workspaceId, workflow.status, max(workflow.statusLastChangedDate) ... group by workspaceId, workflow.status
-      val workflowDatesQuery = for {
-        submissions <- submissionQuery if submissions.workspaceId.inSetBind(workspaceIds)
-        workflows <- workflowQuery if submissions.id === workflows.submissionId
-      } yield (submissions.workspaceId, workflows.status, workflows.statusLastChangedDate)
+      // submission date query:
+      //
+      // select workspaceId, case when subFailed > 0 then 'Failed' else 'Succeeded' end as status, max(subEndDate) as subEndDate
+      // from (
+      //   select submission.id, submission.workspaceId, max(case when w.status = 'Failed' then 1 else 0 end) as subFailed, max(w.status_last_changed) as subEndDate
+      //   from submission
+      //   join workflow on workflow.submissionId = submission.id
+      //   where submission.workspaceId in (:workspaceIds)
+      //   and status in ('Succeeded', 'Failed')
+      //   group by 1, 2) v
+      // group by 1, 2
+      //
+      // Explanation:
+      // - inner query gets 2 things per (workspace, submission):
+      // -- whether or not the submission contains any failed workflows
+      // -- the most recent workflow status change date
+      // - outer query gets the most recent workflow status change date per (workspace, submissionStatus), where submissionStatus is:
+      // -- Failed if the submission contains failed workflows
+      // -- Succeeded if the submission does not contain failed workflows
+
+      val innerSubmissionDateQuery = (
+        for {
+          submissions <- submissionQuery if submissions.workspaceId.inSetBind(workspaceIds)
+          workflows <- workflowQuery if submissions.id === workflows.submissionId
+        } yield (submissions.id, submissions.workspaceId, workflows.status, workflows.statusLastChangedDate)
+      ).filter { case (_, _, status, _) =>
+        status === WorkflowStatuses.Failed.toString || status === WorkflowStatuses.Succeeded.toString
+      }.map { case (submissionId, workspaceId, status, statusLastChangedDate) =>
+        (submissionId, workspaceId, Case If (status === WorkflowStatuses.Failed.toString) Then 1 Else 0, statusLastChangedDate)
+      }.groupBy { case (submissionId, workspaceId, _, _) =>
+        (submissionId, workspaceId)
+      }.map { case ((submissionId, workspaceId), recs) =>
+        (submissionId, workspaceId, recs.map(_._3).max, recs.map(_._4).max)
```

Comment:
Yeah, I meant `sum` -- sorry. This jumped out at me because the `outerSubmissionDateQuery` line calls this value `numFailures`. If it's just a 1/0 then maybe just call `exists` here so the later line can be `hasFailures`.